client/containers/domain-autocomplete/actions.js

```diff
@@ -33,33 +37,74 @@ import {
   DOMAIN_AUTOCOMPLETE_SET_VISITED_DOMAIN_LIST,
 } from './mutation-types';
 import {
-  DOMAIN_AUTOCOMPLETE_VISITED_DOMAIN_LIST,
   DOMAIN_AUTOCOMPLETE_SEARCH_URL,
+  DOMAIN_AUTOCOMPLETE_VISITED_DOMAIN_LIST,
 } from './getter-types';
 import { DEBOUNCE_WAIT } from './constants';
-import { updateVisitedDomainList } from './helpers';
+import {
+  updateVisitedDomainList,
+  filterDuplicatesFromDomainList,
+} from './helpers';
 import { httpService } from '~services';
 
 const actions = {
   [DOMAIN_AUTOCOMPLETE_FETCH_DOMAIN_LIST]: debounce(
     async ({ commit, getters }) => {
       const searchUrl = getters[DOMAIN_AUTOCOMPLETE_SEARCH_URL];
+      const allowedCrossOrigin = getters[CROSS_REGION_ALLOWED_CROSS_ORIGIN];
+      const clusterOriginList = getters[CROSS_REGION_CLUSTER_ORIGIN_LIST];
+
+      const fetchDomainForOrigin = async ({ clusterName = '', origin }) => {
+        try {
+          const result = await httpService.get(
+            `${origin}${searchUrl}`,
+            origin && {
+              credentials: 'include',
+              mode: 'cors',
+            }
+          );
 
-      const domainList = await httpService.get(searchUrl);
+          return result;
+        } catch (error) {
+          console.warn(
+            `Failed to fetch result from cluster "${clusterName}" with origin "${origin}"`
```

Comment:
probably worth adding the error object? (I'm not actually sure what that'll be / if it'll log reasonably tho)

Reply:
In this circumstance we don't want it to blow up the frontend, hence why we catch the error. Also its possible that the domain doesn't exist in the other region (i.e. local domains which are only local to 1 region), so in those cases its fine to just log a warn message to console than to display an error to the user. Hopefully that makes sense?